src/main/java/com/faforever/client/tournament/game/TournamentGameService.java

```diff
@@ -0,0 +1,122 @@
+package com.faforever.client.tournament.game;
+
+import com.faforever.client.fx.JavaFxUtil;
+import com.faforever.client.game.GameService;
+import com.faforever.client.i18n.I18n;
+import com.faforever.client.notification.ImmediateNotification;
+import com.faforever.client.notification.NotificationService;
+import com.faforever.client.notification.Severity;
+import com.faforever.client.remote.FafServerAccessor;
+import com.faforever.client.theme.UiService;
+import com.faforever.commons.lobby.GameLaunchResponse;
+import com.faforever.commons.lobby.IsReadyRequest;
+import com.faforever.commons.lobby.MatchmakerMatchCancelledResponse;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.CompletableFuture;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class TournamentGameService implements InitializingBean {
+  private static final int SAFETY_CLOSE_NOTIFICATION_SECONDS = 200;
+  private final FafServerAccessor fafServerAccessor;
+  private final I18n i18n;
+  private final Timer timer = new Timer(true);
+  private final UiService uiService;
+  private final GameService gameService;
+  private final NotificationService notificationService;
+  private ImmediateNotification notification;
+  private CompletableFuture<Void> matchFuture;
+
+
+  @Override
+  public void afterPropertiesSet() throws Exception {
+    fafServerAccessor.addEventListener(IsReadyRequest.class, this::onReadRequest);
+    fafServerAccessor.addEventListener(MatchmakerMatchCancelledResponse.class, this::onMatchCanceled);
+    fafServerAccessor.addEventListener(GameLaunchResponse.class, this::onGameLaunch);
+  }
+
+  private void onGameLaunch(GameLaunchResponse gameLaunchResponse) {
+    dismissNotification();
+  }
+
+  private void onMatchCanceled(MatchmakerMatchCancelledResponse matchmakerMatchCancelledResponse) {
+    dismissNotification();
+    cancelMatch();
+    notificationService.addImmediateInfoNotification(
+        "isReady.matchCanceled"
+    );
+  }
+
+  private void onReadRequest(IsReadyRequest isReadyRequest) {
+    log.info("Tournament game is ready, asking user.");
+    if (notification != null) {
+      log.warn("Tournament ready request ignored because tournament is already in progress.");
+      respondToReadyRequest(isReadyRequest.getRequestId(), isReadyRequest);
+      return;
+    }
+    uiService.bringMainStageToFront();
+    final var controller = initializeIsReadyController(isReadyRequest);
+    notification =
+        new ImmediateNotification(i18n.get("isReady.title"), i18n.get("isReady.message", isReadyRequest.getGameName()),
+            Severity.INFO, null, List.of(), controller.getRoot(), false);
+    notificationService.addNotification(notification);
```

Comment:
maybe add a new NotificationService Method instead of creating it if this is common?

Reply:
This is the only point we do such a notification that can not be canceled. Or what do you mean?

Reply:
ah didn't see the not cancellable nature of it. Although still might be good to make it easier to use in the future but not necessary I suppose